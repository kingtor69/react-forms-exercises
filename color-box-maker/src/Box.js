import React, {useState} from 'react';

const Box = ({ color, width, height, id, removeBox }) => {
  const testId = `box${id}`;
  const removeThis = () => { removeBox(id) };
  return (
    <div className="Box">
      <div 
        className="boxDiv"
        style={{
          backgroundColor: color,
          width: `${width}px`,
          height: `${height}px`
        }}
        data-testid={testId}
      >
      </div>
      <button 
        className="removeBoxButton"
        onClick={removeThis}
      >
        X
      </button>
    </div>
  )
}

export default Box;
