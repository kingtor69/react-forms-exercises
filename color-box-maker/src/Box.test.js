import { render } from '@testing-library/react';
import Box from './Box';

it('renders witout ceashing', () => {
  render(<Box />);
});

it('renders snapshot', () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders a new box', () => {
  const newBox = {
    color: "red",
    width: "90",
    height: "150px"
  };
  const { getByTestId, getByText } = 
    render(<Box 
      color={newBox.color}
      width={newBox.width}
      height={newBox.height}
      id={0}
    />);

  const newBoxDiv = getByTestId("box0");
  expect(newBoxDiv).toBeInTheDocument();
  expect(newBoxDiv.className).toBe('boxDiv');
  expect(newBoxDiv.style.backgroundColor).toBe("red");
});
