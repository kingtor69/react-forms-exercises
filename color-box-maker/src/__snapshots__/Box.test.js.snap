// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="boxDiv"
      data-testid="boxundefined"
    />
    <button
      class="removeBoxButton"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
